src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetExpansionClient.cs

```diff
@@ -393,7 +452,317 @@ public virtual bool TryInsertExpansion(int startPositionInSubjectBuffer, int end
+        {
+            AssertIsForeground();
+
+            if (ExpansionSession is null)
+            {
+                return;
+            }
+
+            if (SymbolEqualityComparer.Default.Equals(_state._symbol, symbol))
+            {
+                return;
+            }
+
+            var symbolName = _state._symbol?.Name ?? _state._symbols.FirstOrDefault()?.Name;
+            if (symbolName != symbol.Name)
+            {
+                // Can't track this request
+                _state.ClearSymbolInformation();
+                return;
+            }
+
+            if (!_state._symbols.IsDefaultOrEmpty && _state._symbols != symbols)
+            {
+                // Can't track this request
+                _state.ClearSymbolInformation();
+                return;
+            }
+
+            var textViewModel = TextView.TextViewModel;
+            if (textViewModel == null)
+            {
+                Debug.Assert(TextView.IsClosed);
+                return;
+            }
+
+            var buffer = EditorAdaptersFactoryService.GetBufferAdapter(textViewModel.DataBuffer);
+            if (buffer is not IVsExpansion expansion)
+            {
+                return;
+            }
+
+            // Track current argument values
+            if (_state._symbol is not null)
+            {
+                foreach (var previousParameter in _state._symbol.Parameters)
+                {
+                    if (ExpansionSession.GetFieldValue(previousParameter.Name, out var previousValue) == VSConstants.S_OK)
+                    {
+                        _state._arguments = _state._arguments.SetItem(previousParameter.Name, previousValue);
+                    }
+                }
+            }
+
+            var textSpan = new VsTextSpan[1];
+            if (ExpansionSession is null || ExpansionSession.GetSnippetSpan(textSpan) != VSConstants.S_OK)
+            {
+                return;
+            }
+
+            var adjustedTextSpan = textSpan[0];
+            var firstField = _state._symbol?.Parameters.FirstOrDefault()?.Name ?? "placeholder";
+            if (ExpansionSession.GetFieldSpan(firstField, textSpan) != VSConstants.S_OK)
+            {
+                return;
+            }
+
+            adjustedTextSpan.iStartLine = textSpan[0].iStartLine;
+            adjustedTextSpan.iStartIndex = textSpan[0].iStartIndex;
+
+            var lastField = _state._symbol?.Parameters.LastOrDefault()?.Name ?? "placeholder";
+            if (ExpansionSession.GetFieldSpan(lastField, textSpan) != VSConstants.S_OK)
+            {
+                return;
+            }
+
+            adjustedTextSpan.iEndLine = textSpan[0].iEndLine;
+            adjustedTextSpan.iEndIndex = textSpan[0].iEndIndex;
+
+            var snippet = CreateSnippet(symbol.Name, includeMethod: false, symbol.Parameters, cancellationToken);
```

Comment:
can you doc the purpose of `includeMethod`?  i don't really understand it.

Reply:
➡️ It's already documented

Reply:
⏱ updated existing documentation with examples